client/src/components/Upload/UploadBoxMixin.js

```diff
@@ -332,5 +332,9 @@ export default {
             const models = allHids.map((hid) => Galaxy.currHistoryPanel.collection.getByHid(hid));
             return models;
         },
+        getRequestUrl: function (items, history_id) {
+            var data = this.app.toData(items, history_id);
+            return data.fetchRequest ? "/api/tools/fetch" : this.app.uploadPath;
```

Comment:
This seems like it needs to have an appRoot prefix it - e.g. https://github.com/galaxyproject/galaxy/blob/dev/client/src/components/Upload/config.js#L9.